### PR TITLE
Add Couchbase example

### PR DESCRIPTION
**Pull request by PrasadG193**
Signed-off-by: Prasad Ghangal <prasad.ghangal@gmail.com>

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [x] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: examples/stable/couchbase/couchbase-blueprint.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            rm -rf /backups/test
+            mkdir /backups/test
+            cbbackupmgr config --archive /backups/test --repo couchbase
+            cbbackupmgr backup                                              \
+              --archive /backups/test                                       \
+              --repo couchbase                                              \
+              --cluster couchbase://{{ .Object.metadata.name }}.{{ .Object.metadata.namespace }}.svc    \
+              --username {{ .Phases.TakeBackup.Secrets.authSecret.Data.username | toString }} \
+              --password {{ .Phases.TakeBackup.Secrets.authSecret.Data.password | toString }}
+            echo "tar zcvf - -C /backups/ test | kando location push --profile '{{ toJson .Profile }}' --path '/couchbase-backups/{{ toDate "2006-01-02T15:04:05.999999999Z07:00" .Time  | date "2006-01-02T15-04-05" }}/backup.tgz' -"
```
Comment (PrasadG193) [inline examples/stable/couchbase/couchbase-blueprint.yaml]: Remove echo command

File: examples/stable/couchbase/couchbase-blueprint.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    - func: PrepareData
+      name: TakeBackup
+      objects:
+        authSecret:
+          kind: Secret
+          name: "{{ .Object.spec.authSecret }}"
+          namespace: "{{ .Object.metadata.namespace }}"
+      args:
+        namespace: "{{ .Object.metadata.namespace }}"
+        image: prasadg193/cb-kanister-tools:0.21.0
```
Comment (PrasadG193) [inline examples/stable/couchbase/couchbase-blueprint.yaml]: Use correct image

File: examples/stable/couchbase/couchbase-blueprint.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            rm -rf /backups/test
+            mkdir -p /backups/test
+            cbbackupmgr config --archive /backups/test --repo couchbase
+            cbbackupmgr backup                                              \
+              --archive /backups/test                                       \
+              --repo couchbase                                              \
+              --cluster couchbase://{{ .Object.metadata.name }}.{{ .Object.metadata.namespace }}.svc    \
+              --username {{ .Phases.TakeBackup.Secrets.authSecret.Data.username | toString }} \
+              --password {{ .Phases.TakeBackup.Secrets.authSecret.Data.password | toString }}
+            tar zcvf - -C /backups/ test | kando location push --profile '{{ toJson .Profile }}' --path '/couchbase-backups/{{ toDate "2006-01-02T15:04:05.999999999Z07:00" .Time  | date "2006-01-02T15-04-05" }}/backup.tgz' -
```
Comment (viveksinghggits) [inline examples/stable/couchbase/couchbase-blueprint.yaml]: @PrasadG193 
is the space between `/backups/` and `test` is expected for the command
```
tar zcvf - -C /backups/ test 
```

File: examples/stable/couchbase/README.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
+# Couchbase
+
+[Couchbase](https://www.couchbase.com) is an open-source, distributed multi-model NoSQL document-oriented database software package that is optimized for interactive applications. Couchbase Server is designed to provide easy-to-scale key-value or JSON document access with low latency and high sustained throughput. It is designed to be clustered from a single machine to very large-scale deployments spanning many machines.
+
+## Prerequisites
+
+* Kubernetes 1.9+
+* PV support on the underlying infrastructure
+* Kanister controller version 0.22.0 installed in your cluster
```
Comment (tdmanv) [inline examples/stable/couchbase/README.md]:
```suggestion
* Kanister controller version 0.23.0 installed in your cluster
```

**Pull request merged**